sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.cs

```diff
@@ -322,16 +322,22 @@ private static int ParseHealthcareEntityIndex(string reference)
 
         internal static PiiTask ConvertToPiiTask(RecognizePiiEntitiesAction action)
         {
+            var parameters = new PiiTaskParameters()
+            {
+                Domain = action.DomainFilter.GetString() ?? (PiiTaskParametersDomain?)null,
+                ModelVersion = action.ModelVersion,
+                StringIndexType = Constants.DefaultStringIndexType,
+                LoggingOptOut = action.DisableServiceLogs
+            };
+
+            if (action.CategoriesFilter.Count > 0)
```

Comment:
make sure we don't send empty categories filter if the user didn't set it but the SDK initialized the list

Reply:
I'm not super familiar with the code so I may be lacking context, but if you initialize it with a `ChangeTrackingList`, it should already handle this for you.

Reply:
`action.CategoriesFilter` is defined here: https://github.com/Azure/azure-sdk-for-net/blob/feature/textanalyticsv31/sdk/textanalytics/Azure.AI.TextAnalytics/src/RecognizePiiEntitiesAction.cs#L53

The one that actually gets serialized is this one https://github.com/Azure/azure-sdk-for-net/blob/feature/textanalyticsv31/sdk/textanalytics/Azure.AI.TextAnalytics/src/Internal/PiiTaskParameters.cs#L16.

I tried to make `RecognizePiiEntitiesAction.CategoriesFilter` a `ChangeTrackingList` but the `PiiTaskParameters.CategoriesFilter` property always got serialized as empty :| . I played with it and eventually gave up.

I can keep trying if you it will make a difference

Reply:
Funny, not sure what's going on. But no worries. If it behaves as intended, no complaints from my side.